src/main/java/Duke/System/Parser.java

```diff
@@ -0,0 +1,121 @@
+package Duke.System;
+
+import Duke.Commands.Command;
+import Duke.Commands.CommandClear;
+import Duke.Commands.CommandDeadline;
+import Duke.Commands.CommandDelete;
+import Duke.Commands.CommandEvent;
+import Duke.Commands.CommandExit;
+import Duke.Commands.CommandFind;
+import Duke.Commands.CommandHelp;
+import Duke.Commands.CommandList;
+import Duke.Commands.CommandMark;
+import Duke.Commands.CommandToDo;
+import Duke.Commands.CommandUnmark;
+import Duke.DukeException.DukeException;
+import Duke.Enums.Commands;
+
+public class Parser {
+    public Parser() {}
+
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] params = fullCommand.split("/");
+
+        if (params.length == 0) {
+            throw new DukeException("Invalid input!");
+        }
+
+        String command = params[0].toUpperCase();
+
+        boolean containsCommand = false;
+        int commandNum = -1;
+        Commands[] commands = Commands.values();
+        for (Commands c : commands) {
+            if (c.toString().equals(command)) {
+                commandNum = c.ordinal();
+                containsCommand = true;
+                break;
+            }
+        }
+
+        if (!containsCommand) {
+            throw new DukeException("Invalid command!");
+        }
+
+        return select(commandNum, params);
+    }
+
+    public static Command select(int num, String[] params) throws DukeException {
```

Comment:
You could make the function name a bit more informational . Like making it selectTask instead.